ufo/key_distributor.py

```diff
@@ -44,9 +44,34 @@ def _distribute_key(self, proxy_server, key_string):
     """
     app.logger.info('Distributing keys to proxy server: %s', proxy_server.name)
     client = ssh_client.SSHClient()
-    client.connect(proxy_server)
 
-    # TODO do stuff
+    try:
+      client.connect(proxy_server)
+    except (ssh_client.SSHConnectionException, 
+            ssh_client.InvalidKeyTypeException):
+      app.logger.error(
+          'Unable to connect to proxy server to distribute keys: %s',
```

Comment:
Maybe move the %s string by the proxy server so it doesn't look like we're writing the keys out to the logs. Obviously, you are just writing the proxy server's name, but it reads kind of funny here.